codepipeline.tf

```diff
@@ -1,6 +1,8 @@
 // IAM
 resource "aws_iam_role" "default" {
-  name = "CodePipelineRole"
+  count = var.deployment_controller_type == "CODE_DEPLOY" ? 1 : 0
+
+  name = "${module.this.name}-codepipeline"
```

Comment:
`module.this.name` will just be like `dotcom` or whatever, so deploying across multiple regions will cause the names to collide. `module.this.id` would be preferable.